### ISSUE TITLE
[Bug]: TA 3.0.4 crashes with ATEM Extreme Iso

### ISSUE THREAD

**Issue by Fred-DTV** [josephdadams/TallyArbiter#478]
### What happened?

Unfortunately the current TA Pre-Release does find my AEI automatically but as soon as i add it to the sources TA crashes.

Any idea what could cause this?
I've done a full reinstall already

### Version

3.0.4

### Distribution

Desktop

### OS

Windows

### TallyArbiter configuration

```javascript
{
  "security": {
    "jwt_private_key": "u/EhgpRu0DmpD+iiV1yT9uqHqt1JAeKPVrUGZJBoVjoug5Lg0qE22GHRdxbWUHU8CD+kSW1KOe1cuf7PpQTjWHmVFy2OWh339hRC/4D19R5MMOIRmrBtoeDdJp/3AD3CyQjMZDUs/S4saW8XOXH8WNERK3rytgxYAk9BfEgWuUDOVNTepVZ9ZJKZ+5F9o/KaSeB4aJdp/glpmPx5SCQPMvCkBcnsWFHagHvj4CmrTRQTMiCjTHFRsSGGmdpffobzsRY0AHppE4JWTkauurAurB7Azrg28TV1hvhO36i1WcJM1MKKdfBoHovoUIwrGys21stRFK32l/8nfdcb58Ir0A=="
  },
  "users": [
    {
      "username": "producer",
      "password": "$2b$10$jnEOdmkfY/9qJmc/zs7j5..VBcXdqZNDnRAOVK8LhEHu1MCoHaZSC",
      "roles": "producer"
    },
    {
      "username": "admin",
      "password": "$2b$10$CH7AJV/icndUSGu9VpcTiOqn9iNP/kqAmh4vpE5YbWMVbqN3SZbD6",
      "roles": "admin"
    }
  ],
  "cloud_destinations": [],
  "cloud_keys": [],
  "device_actions": [],
  "device_sources": [],
  "devices": [],
  "sources": [
    {
      "name": "ATEM Mini Extreme ISO",
      "data": {
        "addresses": [
          "2.39.2.60",
          "ATEM Mini Extreme ISO._blackmagic._tcp.local"
        ],
        "ip": "ATEM Mini Extreme ISO._blackmagic._tcp.local",
        "cut_bus_mode": "on"
      },
      "sourceTypeId": "44b8bc4f",
      "reconnect": true,
      "enabled": true,
      "id": "2d793c28",
      "max_reconnects": 5,
      "connected": false
    }
  ],
  "tsl_clients": [],
  "tsl_clients_1secupdate": false,
  "bus_options": [
    {
      "id": "e393251c",
      "label": "Preview",
      "type": "preview",
      "color": "#00ff00",
      "priority": 50
    },
    {
      "id": "334e4eda",
      "label": "Program",
      "type": "program",
      "color": "#ff0000",
      "priority": 200
    },
    {
      "id": "12c8d699",
      "label": "Aux 1",
      "type": "aux",
      "color": "#0000FF",
      "priority": 100
    },
    {
      "id": "12c8d689",
      "label": "Aux 2",
      "type": "aux",
      "color": "#0000FF",
      "priority": 100
    }
  ],
  "externalAddress": "http://0.0.0.0:4455/#/tally",
  "remoteErrorReporting": false,
  "uuid": "1c477a4d"
}
```

### Error stacktrace (if applicable)

```shell
TypeError: data.hasOwnProperty is not a function
    at _deconstructPacket (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\binary.js:39:22)
    at _deconstructPacket (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\binary.js:40:32)
    at _deconstructPacket (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\binary.js:40:32)
    at _deconstructPacket (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\binary.js:32:26)
    at _deconstructPacket (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\binary.js:32:26)
    at Object.deconstructPacket (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\binary.js:16:17)
    at Encoder.encodeAsBinary (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\index.js:80:41)
    at Encoder.encode (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io-parser\dist\index.js:42:29)
    at Client._packet (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io\dist\client.js:160:28)
    at Socket.packet (C:\Users\DistilleryStation\AppData\Local\Programs\tallyarbiter\resources\app.asar\node_modules\socket.io\dist\socket.js:160:21)
Error datetime
2021-10-18 15:30:40
Logs:
[18/10/2021, 15:22:41] Loading the stored Tally Arbiter configuration file.
[18/10/2021, 15:22:41] Tally Arbiter Sources loaded.
[18/10/2021, 15:22:41] 2 Sources configured.
[18/10/2021, 15:22:41] Tally Arbiter Devices loaded.
[18/10/2021, 15:22:41] 4 Devices configured.
[18/10/2021, 15:22:41] Tally Arbiter Device Sources loaded.
[18/10/2021, 15:22:41] Tally Arbiter Device Actions loaded.
[18/10/2021, 15:22:41] Tally Arbiter TSL Clients loaded.
[18/10/2021, 15:22:41] TSL Clients 1 Second Updates are turned off.
[18/10/2021, 15:22:41] Tally Arbiter Cloud Destinations loaded.
[18/10/2021, 15:22:41] Tally Arbiter Cloud Keys loaded.
[18/10/2021, 15:22:41] Initiating Setup for Source: AMP Iso. Type: Blackmagic ATEM
[18/10/2021, 15:22:41] Source: AMP Iso  Creating ATEM Connection.
[18/10/2021, 15:22:41] Initiating Setup for Source: OBS. Type: OBS Studio
[18/10/2021, 15:22:41] Source: OBS  Creating OBS Websocket connection.
[18/10/2021, 15:22:41] Source Setup Complete.
[18/10/2021, 15:22:41] Initializing Device States.
[18/10/2021, 15:22:41] Device States Initialized.
[18/10/2021, 15:22:41] Setting up the REST API.
[18/10/2021, 15:22:41] REST API Setup Complete.
[18/10/2021, 15:22:41] Starting socket.IO Setup.
[18/10/2021, 15:22:41] Socket.IO Setup Complete.
[18/10/2021, 15:22:41] Starting OSC Setup.
[18/10/2021, 15:22:41] Starting VMix Emulation Service.
[18/10/2021, 15:22:41] Starting HTTP Server.
[18/10/2021, 15:22:42] OSC Sending Port Ready. Broadcasting on Port: 5958
[18/10/2021, 15:22:42] Finished VMix Emulation Setup. Listening for VMix Tally Connections on TCP Port 8099.
[18/10/2021, 15:22:42] Tally Arbiter running on port 4455
[18/10/2021, 15:22:42] Source: OBS  OBS Connection opened.
[18/10/2021, 15:22:42] OBS Tally Source: 5b41deb0 Added new source: {{STREAMING}}
[18/10/2021, 15:22:42] OBS Tally Source: 5b41deb0 Added new source: {{RECORDING}}
[18/10/2021, 15:22:42] Source: OBS  OBS Authenticated.
[18/10/2021, 15:23:40] Source: AMP Iso  Creating ATEM Connection.
[18/10/2021, 15:23:44] Source Edited: AMP Iso
[18/10/2021, 15:23:44] Source: AMP Iso  Creating ATEM Connection.
[18/10/2021, 15:23:44] Config file saved to disk.
[18/10/2021, 15:27:00] Source: AMP Iso ATEM connection opened.
[18/10/2021, 15:27:00] Source: AMP Iso ATEM connection opened.
[18/10/2021, 15:27:00] Source: AMP Iso ATEM connection opened.
[18/10/2021, 15:27:39] Source Added: ndi?
[18/10/2021, 15:27:39] Source: ndi?  Creating VMix connection.
[18/10/2021, 15:27:39] Config file saved to disk.
[18/10/2021, 15:27:39] Source: ndi?  VMix Connection Opened.
[18/10/2021, 15:27:39] VMix Tally Source: ab7ef883 Added new source: Recording
[18/10/2021, 15:27:39] VMix Tally Source: ab7ef883 Added new source: Streaming
[18/10/2021, 15:27:39] New VMix Emulator Connection from ::ffff:2.39.2.99:50146
[18/10/2021, 15:27:39] VMix Emulator Connection ::ffff:2.39.2.99:50146 subscribed to tally
[18/10/2021, 15:27:39] Source: ndi?  VMix data received.
[18/10/2021, 15:28:12] Source: ndi?  Closing VMix connection.
[18/10/2021, 15:28:12] Source Deleted: ndi?
[18/10/2021, 15:28:12] Config file saved to disk.
[18/10/2021, 15:28:12] VMix Emulator Connection ::ffff:2.39.2.99:50146 unsubscribed to tally
[18/10/2021, 15:28:12] VMix Emulator Connection from ::ffff:2.39.2.99:50146 closed
[18/10/2021, 15:28:12] Source: ndi?  VMix Connection closed.
[18/10/2021, 15:28:23] TSL Client Added: 2.39.2.151:undefined (tcp)
[18/10/2021, 15:28:23] TSL Client: c3a1fa14  Initiating TSL Client TCP Socket.
[18/10/2021, 15:28:23] Caught exception: RangeError [ERR_SOCKET_BAD_PORT]: Port should be >= 0 and < 65536. Received NaN.
[18/10/2021, 15:30:30] TSL Client Added: 2.39.2.151:5961 (tcp)
[18/10/2021, 15:30:30] TSL Client: a7991f8d  Initiating TSL Client TCP Socket.
[18/10/2021, 15:30:30] Config file saved to disk.
[18/10/2021, 15:30:30] TSL Client a7991f8d Connection Established: 2.39.2.151:5961
[18/10/2021, 15:30:30] Caught exception: TypeError: data.hasOwnProperty is not a function
[18/10/2021, 15:30:37] TSL Client: c3a1fa14  Closing TSL Client TCP Socket.
[18/10/2021, 15:30:37] TSL Client Deleted: 2.39.2.151:undefined (tcp)
[18/10/2021, 15:30:37] Config file saved to disk.
[18/10/2021, 15:30:37] Caught exception: TypeError: data.hasOwnProperty is not a function
[18/10/2021, 15:30:40] Caught exception: TypeError: data.hasOwnProperty is not a function
Config file:
{
  "externalAddress": "http://0.0.0.0:4455/#/tally",
  "security": {
    "username_settings": "admin",
    "password_settings": "12345",
    "username_producer": "producer",
    "password_producer": "12345"
  },
  "sources": [
    {
      "data": {
        "ip": "2.39.2.60",
        "me_onair": [
          "1",
          "2",
          "3",
          "4"
        ],
        "cut_bus_mode": "off"
      },
      "name": "AMP Iso",
      "sourceTypeId": "44b8bc4f",
      "reconnect": true,
      "enabled": true,
      "id": "dbb43182",
      "connected": true
    },
    {
      "data": {
        "ip": "127.0.0.1",
        "port": 4444,
        "password": "dachsundwiesel"
      },
      "name": "OBS",
      "sourceTypeId": "4eb73542",
      "reconnect": true,
      "enabled": true,
      "id": "5b41deb0",
      "connected": true
    }
  ],
  "devices": [
    {
      "name": "Main Cam",
      "description": "BMD 6k",
      "tslAddress": 1,
      "enabled": true,
      "id": "d0b2d022"
    },
    {
      "name": "Close Up Cam",
      "description": "Birddog",
      "tslAddress": 2,
      "enabled": true,
      "id": "10df4ec3"
    },
    {
      "name": "Guest Cam",
      "description": "BMD 4k",
      "tslAddress": 3,
      "enabled": true,
      "id": "50201827"
    },
    {
      "name": "DJ Cam",
      "description": "Lumix G85",
      "tslAddress": 4,
      "enabled": true,
      "id": "05138ef3"
    }
  ],
  "device_sources": [
    {
      "deviceId": "d0b2d022",
      "sourceIdx": "0",
      "address": "1",
      "sourceId": "dbb43182",
      "id": "21d3e0e9"
    },
    {
      "deviceId": "10df4ec3",
      "sourceIdx": "0",
      "address": "2",
      "rename": false,
      "sourceId": "dbb43182",
      "id": "f1282753"
    },
    {
      "deviceId": "50201827",
      "sourceIdx": "0",
      "address": "3",
      "rename": false,
      "sourceId": "dbb43182",
      "id": "cd1ef5bf"
    },
    {
      "deviceId": "05138ef3",
      "sourceIdx": "0",
      "address": "4",
      "rename": false,
      "sourceId": "dbb43182",
      "id": "8fe5b35b"
    }
  ],
  "device_actions": []
}
```

**Comment by MSP1978** [comment]
The same here!

**Comment by Jocke4f** [comment]
Does this work on TA 3.0.5 or is it still an issue?

**Comment by Jocke4f** [comment]
I borrowed an AEI and tested it both with v3.0.6 and v3.0.7. It worked fine to automatically discover and add the AEI using the IP address. Based on this do I suggest that we close this based on the ATEM fix that was included in v3.0.5.
